include/zephyr/kernel.h

```diff
@@ -946,6 +946,23 @@ __syscall void k_thread_priority_set(k_tid_t thread, int prio);
 __syscall void k_thread_deadline_set(k_tid_t thread, int deadline);
 #endif
 
+/**
+ * @brief Invoke the scheduler
+ *
+ * This routine invokes the scheduler to force a schedule point. If invoked
+ * from within a thread, the scheduler will be invoked immediately (provided
+ * interrupts were not locked when invoked). If invoked from within an ISR,
+ * the scheduler will be invoked upon exiting the ISR.
+ *
+ * Invoking the scheduler allows the kernel to make an immediate determination
+ * as to what the next thread to execute should be. This is not the same as
+ * yielding for that voluntarily surrenders control of the CPU to another
+ * thread of equal or greater priority.
```

Comment:
Might need more clarity here.  The distinction the reader wants to understand is that yield will always choose a different thread of the same priority to run, and this won't.